hudi-flink/src/main/java/org/apache/hudi/sink/bootstrap/BootstrapOperator.java

```diff
@@ -211,16 +213,29 @@ protected void loadRecords(String partitionPath) throws Exception {
             return;
           }
 
-          final List<HoodieKey> hoodieKeys;
-          try {
-            hoodieKeys =
-                fileUtils.fetchRecordKeyPartitionPath(this.hadoopConf, new Path(baseFile.getPath()));
-          } catch (Exception e) {
-            throw new HoodieException(String.format("Error when loading record keys from file: %s", baseFile), e);
-          }
-
-          for (HoodieKey hoodieKey : hoodieKeys) {
-            output.collect(new StreamRecord(new IndexRecord(generateHoodieRecord(hoodieKey, fileSlice))));
+          if (this.batchSize > 0) {
+            List<HoodieKey> hoodieKeys;
+            Path filePath = new Path(baseFile.getPath());
+            try (BaseFileUtils.BaseFileReader reader = fileUtils.getReader(this.hadoopConf, filePath)) {
+              do {
+                hoodieKeys = fileUtils.fetchRecordKeyPartitionPath(reader, filePath, batchSize);
```

Comment:
1. Can we have method `BaseFileReader#hasNextBatch` and `BaseFileReader#NextBatch` then iterator over that ?
2. Can we also have a constant batch size `1024` and remove the config option then ?